File: Graph Algorithms in Genome Sequencing/Genome seq/Reconstruct a String from its k-mer Composition/StringReconstruction.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class StringReconstruction {
    /*
     * Reconstructs a string from its k-mer composition
     *
     * @param k         the length of the k-mers in the composition
     * @param patterns  a List of Strings containing the k-mer composition
     *
     * @return a String text with k-mer composition equal to patterns
     */
    public static String reconstruct(int k, List<String> patterns) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);

        int k = stdin.nextInt();
        stdin.nextLine(); // ignore newline char after k

        List<String> patterns = new ArrayList<>();

        while (stdin.hasNextLine()) patterns.add(stdin.nextLine());

        System.out.println(reconstruct(k, patterns));
    }
}

File: Graph Algorithms in Genome Sequencing/Genome seq/Reconstruct a String from its k-mer Composition/string_reconstruction.cpp
#include <iostream>
#include <string>
#include <vector>


using std::cout;
using std::cin;
using std::endl;
using std::string;
using std::vector;


/*
 * Reconstructs a string from its k-mer composition
 *
 * \param k         the length of the k-mers in the composition
 * \param patterns  a vector of strings containing the k-mer composition
 *
 * \return a string text with k-mer composition equal to patterns
 */
string reconstruct(int k, const vector<string>& patterns) {
    // TODO: your code here
    return "";
}


int main(int argc, char* argv[]) {
    int k;
    cin >> k;

    vector<string> patterns;
    string pattern;

    while (cin >> pattern) {
        patterns.push_back(pattern);
    }

    cout << reconstruct(k, patterns) << endl;
}
